client/src/components/survey/survey.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import { connect } from 'react-redux'
+// Helpers
+import classNames from 'classnames/bind'
+import apiCall from '../../api-call'
+import { loadQuestions } from '../../actions'
+// Components
+import Question from './question'
+import Answer from './answer'
+// Styles
+import styles from './survey.scss'
+
+const css = classNames.bind(styles)
+
+const mapStateToProps = state => ({
+  questions: state.questions,
+})
+
+class Survey extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.nextQuestion = this.nextQuestion.bind(this)
+    this.previousQuestion = this.previousQuestion.bind(this)
+
+    this.state = {
+      currentIndex: 0,
+    }
+  }
+
+  async componentDidMount() {
+    const questions = await apiCall('GET', '/questions')
+    this.props.dispatch(loadQuestions(questions))
+  }
+
+  nextQuestion() {
+    this.setState(state => ({
+      ...state,
+      currentIndex: Math.min(this.props.questions.length - 1, state.currentIndex + 1),
+    }))
+  }
+
+  previousQuestion() {
+    this.setState(state => ({
+      ...state,
+      currentIndex: Math.max(0, state.currentIndex - 1),
+    }))
+  }
+
+  buttons() {
+    const { currentIndex } = this.state
+    const { questions } = this.props
+    const hasNext = currentIndex < questions.length - 1
+    const hasPrevious = currentIndex > 0
+    return (
+      <div className={css('button-container')}>
+        {
+          hasPrevious ? (
+            <button type="button" className={css('previous-question')} onClick={this.previousQuestion}>
+              Previous
+            </button>
+          ) : null
+        }
+        {
+          hasNext ? (
+            <button type="button" className={css('next-question')} onClick={this.nextQuestion}>
+              Next
+            </button>
+          ) : null
+        }
+      </div>
+    )
+  }
+
+  render() {
+    const { currentIndex } = this.state
+    const { questions } = this.props
+    if (questions.length < 1) {
+      // NOT UNNECESSARY STATE
```

Comment:
can be removed as we disable the button if ther is not state

Reply:
can not be removed, if it's removed the app will crash as the survey panel is still mounted, just not displayed